javascript/script.js

```diff
@@ -0,0 +1,200 @@
+// End of global variables
+
+
+
+// Function to update HTML elements with global variables created through info. retrieved from NWS API
+function update_weather_info() {
+    // clears previous HTML content so that it works for multiple usage 
+    document.getElementById('forecast_temperature_content').innerHTML = '';
+    document.getElementById('forecast_precipitation_content').innerHTML = '';
+    document.getElementById('forecast_wind_content').innerHTML = '';
+    document.getElementById('forecast_text_content').innerHTML = '';
+
+    // Makes sure high temp/low temp are in correct indexes (high must be in 0, low must be in 1)
+    if (temperature[1] > temperature[0]) {
+        let low_temp = temperature[0];
+        temperature[0] = temperature[1];
+        temperature[1] = low_temp;
+    }
+
+    // updates corresponding HTML divs with weather information by using .innerHTML to write new HTML
+    document.getElementById('forecast_temperature_content').innerHTML = `<p>High: ${temperature[0]} °F, Low: ${temperature[1]} °F</p>`;
+    if (precipitation !== null){
+        document.getElementById('forecast_precipitation_content').innerHTML = `<p>Precipitation: ${precipitation} %</p>`; 
+    } else {
+        document.getElementById('forecast_precipitation_content').innerHTML = `<p>Precipitation: 0%</p>`; 
+    }
+    document.getElementById('forecast_wind_content').innerHTML = `<p>Wind: ${wind}</p>`;
+    document.getElementById('forecast_text_content').innerHTML = `<p>${text_overview}</p>`;
+
+    // Update or change images depending on highest temperature (index 0) -> 3 conditions either high, moderate, or low
+    if (temperature[0] > 80) {
+        document.getElementById('temp_image').src = filepath_hot_temp;
+    } else if (temperature[0] > 60) {
+        document.getElementById('temp_image').src = filepath_moderate_temp;
+    } else {
+        document.getElementById('temp_image').src = filepath_low_temp;
+    }
+
+    // Precip
+    if (precipitation > 70) {
+        document.getElementById('precip_image').src = filepath_high_precip;
+    } else if (precipitation > 30) {
+        document.getElementById('precip_image').src = filepath_moderate_precip;
+    } else {
+        document.getElementById('precip_image').src = filepath_low_precip;
+    }
+
+    // Wind, Data from API is a string, so I need to extract relevant info (aka windspeed) from that before comparing 
+    // Process = First join array into a single string, then use a function I found from stackoverflow which returns only integer values
+    let wind_string = wind.join(" ");
+    let wind_speed = find_int_in_str(wind_string);
+    if (wind_speed > 25) {
+        document.getElementById('wind_image').src = filepath_high_wind;
+    } else if (wind_speed > 15) {
+        document.getElementById('wind_image').src = filepath_moderate_wind;
+    } else {
+        document.getElementById('wind_image').src = filepath_low_wind;
+    }
+}
```

Comment:
This function is quite long. You might have decomposed it. Good job using resources to supplement your code (and citing them)